src/tools/GraphemeTableGen/Program.cs

```diff
@@ -0,0 +1,579 @@
+		/* cbExtend        | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, Ω /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbRI            | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, Ω /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbPrepend       | */ [0 /* | */, Ω /*  | */, 0 /* | */, 0 /* | */, 0 /*  | */, 0 /*  | */, 0 /*  | */, 0 /*  |  */, 0 /*  |  */, 0 /*   |   */, 0 /*   |    */, 0 /*     | */, 0 /* | */, 0 /* | */],
+		/* cbHangulL       | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, 0 /*  | */, 0 /*  | */, Ω /*  |  */, 0 /*  |  */, 0 /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbHangulV       | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, 0 /*  | */, 0 /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbHangulT       | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, 0 /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbHangulLV      | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, 0 /*  | */, 0 /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbHangulLVT     | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, 0 /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbInCBLinker    | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, Ω /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, 0 /*     | */, Ω /* | */, 0 /* | */],
+		/* cbInCBConsonant | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, Ω /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbExtPic        | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, Ω /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, Ω /* | */, 0 /* | */],
+		/* cbZWJ           | */ [Ω /* | */, Ω /*  | */, 0 /* | */, Ω /* | */, Ω /*  | */, Ω /*  | */, Ω /*  | */, Ω /*  |  */, Ω /*  |  */, Ω /*   |   */, 0 /*   |    */, Ω /*     | */, 0 /* | */, 0 /* | */],
+    ]
+];
+
+if (args.Length != 1)
+{
+    Console.WriteLine(
+        """
+        Usage: GraphemeTableGen <path to ucd.nounihan.grouped.xml>
+
+        You can download the latest ucd.nounihan.grouped.xml from:
+            https://www.unicode.org/Public/UCD/latest/ucdxml/ucd.nounihan.grouped.zip
+        """
+    );
+    Environment.Exit(1);
+}
+
+var ucd = ExtractValuesFromUcd(args[0]);

```

Comment:
the old console tables had support for "overriding" some widths. Are we explicitly ignoring those special ranges with the new measurer?

Reply:
extended note below

Reply:
The override(s) are here: https://github.com/microsoft/terminal/blob/c01f510611fa13bbc116beada2801479fa4d3527/src/tools/GraphemeTableGen/Program.cs#L310-L312

We currently have two more ranges:
```xml
<override first-cp="4DC0" last-cp="4DFF" ea="H" comment="hexagrams are historically narrow" />
<override first-cp="FE20" last-cp="FE2F" ea="H" comment="narrow combining ligatures (split into left/right halves, which take 2 columns together)" />
```

The reason they're missing:
* U+4DC0 to U+4DFF is the Yijing Hexagram Symbols block. The entire block has a `ea="N"` already so the `ea="H"` override doesn't change anything.
* U+FE20 to U+FE2F is the Combining Half Marks block. As the name already indicates, they must be zero-width. If you read the console-style measurement code in `CodepointWidthDetector.cpp`, you'll see that it clamps the final `width` to a value >=1 and so these zero-width codepoints will be turned into the equivalent of `ea="H"` too.

I did test whether my console-style implementation is identical to the current implementation. However, that was before my latest version, and I'd be happy to repeat the tests sometime soon.

Reply:
I approve of this rationale - resolving discussion!